perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -30,7 +28,69 @@ static char const* RegisterNames[][3] =
     {"flags", "flags", "flags"}
 };
 
-static void SimulateInstruction(const instruction& Instruction)
+static char const* FlagNames = { "ODITSZAPC" };
+enum Flags
+{
+    Flag_OF,
+    Flag_DF,
+    Flag_IF,
+    Flag_TF,
+    Flag_SF,
+    Flag_ZF,
+    Flag_AF,
+    Flag_PF,
+    Flag_CF,
+
+    Flag_count
+};
+
+static void SetFlags(u16 Result, bool* Flags)
```

Comment:
Also note how everywhere where you use a variable called `Flags`, you shadow the type of the enum (it is not accessible anymore because Flags refer to your variable in this function).

Because of this, you can't do this for example:
```
Flags MyFlag = Flag_OF;
```

So this is also something to be avoided. I recommend renaming all your variables that are named `Flags`.

Reply:
Great point, fixed
791ab0529796713600a4dd230395ec0b30c85150